app/lib/pdf_filler/nj1040_pdf.rb

```diff
@@ -0,0 +1,111 @@
+        "Text11": spouse_ssn ? spouse_ssn[5] : "",
+        "Text12": spouse_ssn ? spouse_ssn[6] : "",
+        "Text13": spouse_ssn ? spouse_ssn[7] : "",
+        "Text14": spouse_ssn ? spouse_ssn[8] : "",
+
+        # address
+        "SpousesCU Partners SSN if filing jointly": get_address, # address text field
+        "CountyMunicipality Code See Table page 50": @xml_document.at("ReturnHeaderState Filer USAddress CityNm")&.text,  # city / town text field
+        "State": @xml_document.at("ReturnHeaderState Filer USAddress StateAbbreviationCd")&.text,
+        "ZIP Code": @xml_document.at("ReturnHeaderState Filer USAddress ZIPCd")&.text
+      }
+    end
+
+    private
+
+    def get_county_code
+      @xml_document.at("ReturnDataState FormNJ1040 Header CountyCode")&.text
+    end
+
+    def get_taxpayer_ssn
+      @xml_document.at("ReturnHeaderState Filer Primary TaxpayerSSN")&.text
+    end
+
+    def get_address
+      address_line_1 = @xml_document.at("ReturnHeaderState Filer USAddress AddressLine1Txt")&.text
+      address_line_2 = @xml_document.at("ReturnHeaderState Filer USAddress AddressLine2Txt")&.text
+      address_line_1 + (address_line_2.present? ? " #{address_line_2}" : "")
+    end
+
+    def get_name
+      first_name = @xml_document.at("ReturnHeaderState Filer Primary TaxpayerName FirstName")&.text
+      last_name = @xml_document.at("ReturnHeaderState Filer Primary TaxpayerName LastName")&.text
+      middle_initial = @xml_document.at("ReturnHeaderState Filer Primary TaxpayerName MiddleInitial")&.text
+      suffix = @xml_document.at("ReturnHeaderState Filer Primary TaxpayerName NameSuffix")&.text
+
+      spouse_first_name = @xml_document.at("ReturnHeaderState Filer Secondary TaxpayerName FirstName")&.text
+      spouse_last_name = @xml_document.at("ReturnHeaderState Filer Secondary TaxpayerName LastName")&.text
+      spouse_middle_initial = @xml_document.at("ReturnHeaderState Filer Secondary TaxpayerName MiddleInitial")&.text
+      spouse_suffix = @xml_document.at("ReturnHeaderState Filer Secondary TaxpayerName NameSuffix")&.text
+
+      if spouse_first_name.present? && spouse_last_name.present?
+        if last_name == spouse_last_name
+          return "#{format_name(first_name, last_name, middle_initial, suffix)} & #{format_no_last_name(spouse_first_name, spouse_middle_initial, spouse_suffix)}"
+        else
+          return "#{format_name(first_name, last_name, middle_initial, suffix)} & #{format_name(spouse_first_name, spouse_last_name, spouse_middle_initial, spouse_suffix)}"
+        end
+      end
+
+      format_name(first_name, last_name, middle_initial, suffix)
+    end
+
+    def format_name(first_name, last_name, middle_initial, suffix)
+      "#{last_name} #{format_no_last_name(first_name, middle_initial, suffix)}"
+    end
+
+    def format_no_last_name(first_name, middle_initial, suffix)
+      "#{first_name}#{middle_initial ? (" " + middle_initial) : ""}#{suffix ? (" " + suffix) : ""}"
```

Comment:
[DUST] could also do `[first_name, middle_initial, suffix].compact.join(" ")`

Reply:
actually for that matter i guess you could just do `[last_name, first_name, middle_initial, suffix].compact.join(" ")` ?

Reply:
oh jk i now see the need for a separate no-last-name method

Reply:
ok last thing sorry, i just realized i'm not totally sure whether the blank ones will come back as `nil` or empty strings but i was assuming `nil`. in the case of empty strings it would be `.reject(&:empty?)` instead of `.compact`